biogeochem/EDPhysiologyMod.F90

```diff
@@ -2026,7 +2027,7 @@ subroutine recruitment( currentSite, currentPatch, bc_in )
     !----------------------------------------------------------------------
 
     allocate(temp_cohort) ! create temporary cohort
-    call zero_cohort(temp_cohort)
+    call temp_cohort%ZeroValues()
```

Comment:
@adrifoster I've always felt like creating this temporary cohort was kind of an extra unnecessary step that just confuses me each time.  We just use it's data structure's primitives as arguments for the create_cohort routine, which is the routine that generates the real cohort used.  Can we drop the temp_cohort here, and just create a set of the key primitives that are passed to create_cohort() instead?

Reply:
Yes I agree it's not really serving us right now. Unless we wanted to add some kind of checking ability to the cohort type (via setters or getters, etc.), I think it's fine to just use intrinsic types for all of these values. I can add this change.

Reply:
I saw a similar approach in EDInitMod init_cohorts() too btw